compiler/src/dotty/tools/dotc/config/SourceVersion.scala

```diff
@@ -16,6 +16,8 @@ enum SourceVersion:
   // !!! Keep in sync with scala.runtime.stdlibPatches.language !!!
   case `future-migration`, `future`
 
+  case `never`    // needed for MigrationVersion.errorFrom if we never want to issue an error
```

Comment:
should add this to the `illegalSourceVersionNames` list so users can't choose it with `-source:never`